src/app/app.component.html

```diff
@@ -0,0 +1,103 @@
+<div class="container">
+  <form method="post" class="form" [formGroup]="form" (ngSubmit)="onSubmit()">
+    <mat-form-field appearance="fill">
+      <mat-label>Name</mat-label>
+      <input matInput formControlName="firstName" />
+    </mat-form-field>
+
+    <mat-form-field appearance="fill">
+      <mat-label>Last name</mat-label>
+      <input matInput formControlName="lastName" />
+    </mat-form-field>
+
+    <mat-form-field appearance="fill">
+      <mat-label>Choose a date</mat-label>
+      <input matInput [matDatepicker]="picker" formControlName="dateOfBirth" />
+      <mat-hint>MM/DD/YYYY</mat-hint>
+      <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
+      <mat-datepicker #picker></mat-datepicker>
+    </mat-form-field>
+
+    <mat-form-field appearance="fill">
+      <mat-label>FE technology</mat-label>
+      <mat-select formControlName="framework">
+        <mat-option *ngFor="let framework of frameworks" [value]="framework">
+          {{ framework }}
+        </mat-option>
+      </mat-select>
+    </mat-form-field>
+
+    <mat-form-field appearance="fill">
+      <mat-label>FE technology version</mat-label>
+      <mat-select formControlName="frameworkVersion">
+        <mat-option *ngFor="let version of versions" [value]="version">
+          {{ version }}
+        </mat-option>
+      </mat-select>
+    </mat-form-field>
+
+    <mat-form-field appearance="fill">
+      <mat-label>Email</mat-label>
+      <input matInput formControlName="email" />
+      <mat-error *ngIf="email.invalid">{{errorMessage}}</mat-error>
+    </mat-form-field>
+
+    <div class="hobbies__container">
+      <div class="hobbies__title">
+        <h3>Hobbies</h3>
+
+        <button
+          mat-mini-fab
+          color="primary"
+          type="button"
+          class="hobbies__button"
+          (click)="addHobby()"
+        >
+          <mat-icon>add</mat-icon>
+        </button>
+      </div>
+      <ng-container *ngIf="hobbiesForms.length > 0; else empty">
+        <ng-container *ngFor="let hobbyForm of hobbiesForms; let i = index">
+          <div class="hobbies__list">
+            <div [formGroup]="hobbyForm" class="hobbies__list">
+              <mat-form-field
+                appearance="fill"
+                class="hobbies__item"
+                *ngFor="let conf of config"
```

Comment:
if you use config, better to use it in the whole form
if you create fields in template, continue this
please choose one approach and keep it simple